tests/test_utils.py

```diff
@@ -171,3 +174,153 @@ def test_good_common_names(self):
         good_names = ["Human", "Stuff and things"]
         for good_name in good_names:
             self.assertTrue(utils.is_valid_species_common_name(good_name))
+
+
+class TestDownload(unittest.TestCase):
+    def test_download_local_file(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            tmpdir = pathlib.Path(tmpdir)
+            input_filename = tmpdir / "input.foo"
+            output_filename = tmpdir / "output.foo"
+            with open(input_filename, "w") as f:
+                print("foofoofoo", file=f)
+            url = input_filename.resolve().as_uri()
+            utils.download(url, output_filename)
+            self.assertTrue(output_filename.exists())
+            with open(output_filename) as f:
+                contents = f.read()
+            self.assertEqual(contents.strip(), "foofoofoo")
+
+    def test_download_nonexistent(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            tmpdir = pathlib.Path(tmpdir)
+            input_filename = tmpdir / "nonexistant"
+            output_filename = tmpdir / "output.foo"
+            for url in (
+                input_filename.resolve().as_uri(),  # local file
+                "http://example.com/nonexistant",  # remote file
+            ):
+                with self.assertRaises(OSError):
+                    utils.download(url, output_filename)
+                self.assertFalse(output_filename.exists())
+
+
+class TestSha256(unittest.TestCase):
+    def test_sha256(self):
```

Comment:
Be nice to test the chunking functionality here somewhere too. Could we calculate the sha256 on files of varying length using a straight ``f.read()`` and comparing to the lib version?